BackEnd/administration/views.py

```diff
@@ -29,16 +29,47 @@
 from profiles.models import Profile
 from .permissions import IsStaffUser, IsStaffUserOrReadOnly, IsSuperUser
 
+from django_filters.rest_framework import DjangoFilterBackend as filters
+from .filters import UsersFilter
+
 
 class UsersListView(ListAPIView):
     """
     List of users.
+    Query parameters:
+       /?sort=id&direction=asc or /?sort=id&direction=desc
+    Filters:
+       /?company_name= ,/?status=active/inactive/staff/superuser/deleted
     """
 
     permission_classes = [IsStaffUser]
     pagination_class = ListPagination
     serializer_class = AdminUserListSerializer
-    queryset = CustomUser.objects.all().order_by("id")
+    queryset = CustomUser.objects.all()
+    filter_backends = [filters]
+    filterset_class = UsersFilter
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        status = self.request.query_params.get("status", None)
+        if status:
+            if status == "active":
+                queryset = queryset.filter(is_active=True)
+            elif status == "inactive":
+                queryset = queryset.filter(is_active=False)
+            elif status == "staff":
+                queryset = queryset.filter(is_staff=True)
+            elif status == "superuser":
+                queryset = queryset.filter(is_superuser=True)
+            elif status == "deleted":
+                queryset = queryset.filter(email__startswith="is_deleted_")
+
+        sort = self.request.query_params.get("sort", "name")
+        direction = self.request.query_params.get("direction", "asc")
+        if direction == "desc":
+            sort = f"-{sort}"
+        return queryset.order_by(sort)
```

Comment:
There is no need for this method. Filters (django_filters) do all the work.

Reply:
Done